agent/api/container/container.go

```diff
@@ -807,7 +813,17 @@ func (c *Container) MergeEnvironmentVariables(envVars map[string]string) {
 	}
 }
 
-func (c *Container) HasSecretAsEnv() bool {
+//Append the logging driver using secret into the container's LogConfig
+func (c *Container) ApplyLogDriverSecret(hostConfig *dockercontainer.HostConfig, logDriverSecretName string, logDriverSecretValue string) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	hostConfig.LogConfig.Config[logDriverSecretName] = logDriverSecretValue
+}
+
+// MergeLogDrivers appends additional pairs to
+// the the container's Logging Drivers structure
+func (c *Container) HasSecretAsEvnAndLogDriver() bool {
```

Comment:
Think it should be`HasSecretAsEnvOrLogDriver`. either way, just calling it `HasSecrets` or `HasSecretsAvailable` might be better. also the comment needs to be changed.

Reply:
Will change it to HasSecretAsEnvOrLogDriver at this time. It is specifically for checking env and log driver secrets.